analytics/tests/test_counts.py

```diff
@@ -180,3 +180,66 @@ def test_active_users_by_is_bot(self):
         self.assertEqual(InstallationCount.objects.count(), 2)
         self.assertFalse(UserCount.objects.exists())
         self.assertFalse(StreamCount.objects.exists())
+
+    def test_messages_sent_by_message_type(self):
+        # type: () -> None
+        property = 'messages_sent:message_type'
+        stat = COUNT_STATS[property]
+
+        user1 = self.create_user('email1')
+        user2 = self.create_user('email2')
+        user3 = self.create_user('email3')
+
+        private_stream1, rec_private_stream1 = self.create_stream_with_recipient(name='private_stream1', invite_only=True)
+
+        public_stream1, rec_public_stream1 = self.create_stream_with_recipient(name='public_stream1')
+        public_stream2, rec_public_stream2 = self.create_stream_with_recipient(name='public_stream2')
+
+        # 2 private_stream
+        self.create_message(user1, rec_private_stream1)
+        self.create_message(user2, rec_private_stream1)
+
+        # 3 public_stream
+        self.create_message(user1, rec_public_stream1)
+        self.create_message(user2, rec_public_stream1)
+        self.create_message(user3, rec_public_stream1)
+
+        rec_user1 = Recipient.objects.create(type_id=user1.id, type=Recipient.PERSONAL)
+        rec_user2 = Recipient.objects.create(type_id=user2.id, type=Recipient.PERSONAL)
+        rec_user3 = Recipient.objects.create(type_id=user3.id, type=Recipient.PERSONAL)
+
+        huddle1, rec_huddle1 = self.create_huddle_with_recipient('foo')
+        huddle2, rec_huddle2 = self.create_huddle_with_recipient('bar')
+
+        # 5 private_message
+        self.create_message(user1, rec_user2)
+        self.create_message(user2, rec_user1)
+        self.create_message(user3, rec_user3)
+
+        self.create_message(user1, rec_huddle1)
+        self.create_message(user2, rec_huddle2)
+
```

Comment:
This is super thorough, thanks! I was going to suggest adding:
* A message sent 61 minutes ago
* A messages sent 25 hours ago
* A realm with users/huddles/streams but no messages

But I think I'll just add that to setUp for all tests.